Mathlib/Data/Fin/Basic.lean

```diff
@@ -244,6 +244,12 @@ This one instead uses a `NeZero n` typeclass hypothesis.
 theorem val_zero' (n : ℕ) [NeZero n] : ((0 : Fin n) : ℕ) = 0 :=
   rfl
 
+/-- Fin.mk_zero` in `Lean` only applies in `Fin (n + 1)`.
```

Comment:
I don't think this is true: `#check Fin.mk_zero` yields `Fin.mk_zero {n : ℕ} : ⟨0, ⋯⟩ = 0`.

Why can't you use that here (and in the downstream lemmas)?

Reply:
If you go to the source, then you see that LHS and RHS are in `Fin (n + 1)`.

Reply:
Oh right... thanks, sorry for the noise!